lib/process-services-cloud/src/lib/task/task-filters/services/service-task-filter-cloud.service.ts

```diff
@@ -0,0 +1,306 @@
+        const key: string = this.prepareKey(updatedFilter.appName);
+        return this.getTaskFiltersByKey(updatedFilter.appName, key).pipe(
+            switchMap((filters: any) => {
+                if (filters && filters.length === 0) {
+                    return this.createTaskFilters(updatedFilter.appName, key, <ServiceTaskFilterCloudModel[]> [updatedFilter]);
+                } else {
+                    const itemIndex = filters.findIndex((filter: ServiceTaskFilterCloudModel) => filter.id === updatedFilter.id);
+                    filters[itemIndex] = updatedFilter;
+                    return this.updateTaskFilters(updatedFilter.appName, key, filters);
+                }
+            }),
+            map((updatedFilters: ServiceTaskFilterCloudModel[]) => {
+                this.addFiltersToStream(updatedFilters);
+                return updatedFilters;
+            }),
+            catchError((err) => this.handleTaskError(err))
+        );
+    }
+
+    /**
+     * Deletes a task filter
+     * @param filter The filter to delete
+     * @returns Observable of task instance filters without deleted filter
+     */
+    deleteFilter(deletedFilter: ServiceTaskFilterCloudModel): Observable<ServiceTaskFilterCloudModel[]> {
+        const key = this.prepareKey(deletedFilter.appName);
+        return this.getTaskFiltersByKey(deletedFilter.appName, key).pipe(
+            switchMap((filters: any) => {
+                if (filters && filters.length > 0) {
+                    filters = filters.filter(filter => filter.id !== deletedFilter.id);
+                    return this.updateTaskFilters(deletedFilter.appName, key, filters);
+                }
+                return of([]);
+            }),
+            map(filters => {
+                this.addFiltersToStream(filters);
+                return filters;
+            }),
+            catchError((err) => this.handleTaskError(err))
+        );
+    }
+
+    /**
+     * Checks if given filter is a default filter
+     * @param filterName Name of the target task filter
+     * @returns Boolean value for whether the filter is a default filter
+     */
+    isDefaultFilter(filterName: string): boolean {
+        const defaultFilters = this.defaultServiceTaskFilters();
+        return defaultFilters.findIndex((filter) => filterName === filter.name) !== -1;
+    }
+
+    /**
+     * Calls update preference api to update task filter
+     * @param appName Name of the target app
+     * @param key Key of the task filters
+     * @param filters Details of update filter
+     * @returns Observable of updated task filters
+     */
+    private updateTaskFilters(appName: string, key: string, filters: ServiceTaskFilterCloudModel[]): Observable<ServiceTaskFilterCloudModel[]> {
+        return this.preferenceService.updatePreference(appName, key, filters);
+    }
+
+    /**
+     * Gets the username field from the access token.
+     * @returns Username string
+     */
+    getUsername(): string {
```

Comment:
why do we need this instead of importing the service and calling `getCurrentUserInfo` ? seems like useless api that wraps existing api